modules/promotion/src/PromotionStorage.php

```diff
@@ -107,6 +107,46 @@ public function loadByCoupon(OrderTypeInterface $order_type, StoreInterface $sto
   }
 
   /**
+   * Builds a query that will load all promotions that are no longer valid
+   * determined by the base field limiting values.
+   *
+   * @param bool $only_enabled
+   *   Only include currently enabled promotions that have expired.
+   * @return array|bool|\Drupal\Core\Entity\EntityInterface[]
+   *   The expired promotion entities. Returns FALSE if none found.
+   */
+  public function loadExpired($only_enabled = TRUE) {
+    // Subquery to determine the amount of times a coupon has been used.
+    $usage_query = $this->database->select('commerce_promotion_usage', 'cpu');
+    $usage_query->addExpression('COUNT(cpu.promotion_id)', 'count');
+    $usage_query->where('cpu.promotion_id = cpd.promotion_id');
+    $usage_query->groupBy('cpu.promotion_id');
+
+    $query = $this->database->select($this->getDataTable(), 'cpd');
+
+    // We want to get results of queries that have passed their final date or
+    // have met their max usage.
+    $or_condition = $query->orConditionGroup()
```

Comment:
This is using a regular select, when we can run `$this->getQuery()`.

I'd actually rather do a load of `->condition('end_date', gmdate('Y-m-d'), '<')` and `->condition('status', TRUE)`

We need `<` because it's valid if `>=`. This would cancel promotions on their last day.